stays/views.py

```diff
@@ -1,9 +1,62 @@
+from datetime import datetime
+
 from django.views     import View
 from django.http      import JsonResponse
-from django.db.models import Sum
+from django.db.models import Q, F, Count, Sum, Min, Case, When, Subquery, OuterRef
 
 from stays.models    import Staytype, RoomOption
 
+
+class StaytypeListView(View):
+    def get(self, request):
+        category_id = request.GET.get('category', None)
+        location    = request.GET.get('location', None)
+        check_in    = request.GET.get('CheckIn', None)
+        check_out   = request.GET.get('CheckOut', None)
+
+        if not (check_in and check_out):
+            return JsonResponse({"message":"INVALID_DATE"}, status = 400)
+
+        check_in  = datetime.strptime(check_in, '%Y-%m-%d')
+        check_out = datetime.strptime(check_out, '%Y-%m-%d')
+
+        q = Q()
+
+        if category_id:
+            q &= Q(category = category_id) 
+
+        if location:
+            q &= Q(address__icontains = location) 
+
+        stays = Staytype.objects.filter(q)
+
+        if check_in and check_out:
+            q1 = Q(room__book__check_in__range = [check_in, check_out])
+            q2 = Q(room__book__check_out__range = [check_in, check_out])
+            q3 = Q(room__book__check_in__lte = check_in, room__book__check_out__gte = check_out)
+
+            stays = Staytype.objects.filter(q).prefetch_related('room_set', 'room_set__book_set').annotate(
+                total_room_count = Sum('room__quantity'), 
+                booked_room_count = Subquery(
+                    Staytype.objects.annotate(booked_count = Count('room__book__id', filter = (q1|q2|q3))).filter(pk = OuterRef('pk')).values('booked_count')
+                    ), 
+                    is_available = Case(
+                        When(total_room_count = F('booked_room_count'), then = True), 
+                        default = False
```

Comment:
Case문에서 total_room_count가 예약된 room count보다 많을 경우 예약 가능으로 해줘야되는거 아닌가요??

```suggestion
                    is_available = Case(
                        When(total_room_count__gt = F('booked_room_count'), then = True), 
                        default = False
```

Reply:
total_room_count = booked_room_count 일 경우, sold out으로 생각해서 두 값이 같은 조건문으로 작성하였습니다.
이러한 숙소들은 is_available 값이 True가 되고 exclude 하는 방식으로 작성했습니다. 
다시 생각해보니 is_available 필드 자체와 반대되는 의미를 지니게 되네요.
병민님 말씀대로 gt로 수정하여 예약가능한 숙소들은 is_available 값이 True가 되도록 하였습니다.